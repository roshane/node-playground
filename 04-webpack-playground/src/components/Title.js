
const Title = ({ title }) => {
    return (
        <div>
            {title || 'Title of the page'}
        </div>
    )
}

export default Title;
