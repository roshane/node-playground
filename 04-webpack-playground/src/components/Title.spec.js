import TestRenderer from 'react-test-renderer';
import Title from './Title';


describe("<Title/>", () => {

    beforeEach(() => {
        //TODO setup
    });

    test('<Title/> should render default when not provided any title', () => {
        const defaultTitle = 'Title of the page';
        const component = TestRenderer.create(<Title />);
        const instance = component.root;
        expect(instance.props.title).toBe(undefined);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('<Title/> should render the correct text passed as props', () => {
        const currentTitle = 'Hello 123';
        const component = TestRenderer.create(<Title title={currentTitle} />);
        const instance = component.root;
        expect(instance.props.title).toBe(currentTitle);
        expect(component.toJSON()).toMatchSnapshot();
    });
});


