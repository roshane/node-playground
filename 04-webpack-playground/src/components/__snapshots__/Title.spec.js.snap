// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> <Title/> should render default when not provided any title 1`] = `
<div>
  Title of the page
</div>
`;

exports[`<Title/> <Title/> should render the correct text passed as props 1`] = `
<div>
  Hello 123
</div>
`;
